nautobot/docs/plugins/region-and-site-data-migration-guide.md

```diff
@@ -0,0 +1,285 @@
+    location = models.ForeignKey(to="dcim.Location", on_delete=models.PROTECT, blank=True, null=True)
+    name = models.CharField(max_length=20, help_text="The name of this Example.")
+    number = models.IntegerField(default=100, help_text="The number of this Example.")
+    csv_headers = ["name", "number"]
+
+    class Meta:
+        ordering = ["name"]
+...
+```
+
+Make the migration file by running:
+
+```shell
+nautobot-server makemigrations [app_name] -n [migration_name]
+nautobot-server makemigrations example_plugin -n add_location_field_to_example_model
+```
+
+Apply the migration file by running:
+
+```shell
+nautobot-server migrate [app_name]
+nautobot-server migrate example_plugin
+```
+
+## Create an Empty Migration File to Manually Migrate the Data
+
+After you make sure that all `Site`/`Region` related models have `location` fields on them, it is time to migrate `Site`/`Region` Data to `Location`.
+
+We have to create an empty migration file and run an manual migration.
+
+We can do that by creating a migration file first:
+
+```shell
+nautobot-server makemigrations [app_name] -n [migration_file_name] --empty
+nautobot-server makemigrations example_plugin -n migrate_app_data_from_site_to_location --empty
+```
+
+The empty migration file will look like this, the only dependency is our previous migration that added a `location` ForeignKey field to our `ExampleModel`:
+
+```python
+# Generated by Django 3.2.17 on 2023-02-22 15:38
+# 0008_migrate_example_model_data_from_site_to_location
+
+from django.db import migrations
+
+class Migration(migrations.Migration):
+    dependencies = [
+        ("example_plugin", "0007_add_location_field_to_example_model"),
+    ]
+    operations = []
+```
+
+First we need to add a mandatory dependency to the migration file, namely `("dcim", "0030_migrate_region_and_site_data_to_locations")`. This dependent migration is very important as it creates the `Site`/`Region` `LocationType` locations for you to migrate your data to.
+
+**Without it, your manual migration might not work!**
+
+```python
+# Generated by Django 3.2.17 on 2023-02-22 15:38
+# 0008_migrate_example_model_data_from_site_to_location
+
+from django.db import migrations
+
+class  Migration(migrations.Migration):
+    dependencies = [
+        # The dcim migration creates the Site Type and Region Type Locations that
+        # your data models are migrating to. It has to be run **before** this migration.
+        ("dcim", "0030_migrate_region_and_site_data_to_locations"),
+        ("example_plugin", "0007_add_location_field_to_example_model"),
+    ]
+    operations = []
+```
+
+Now we can write the function that does the manual migration. We need to query `ExampleModel` instances that have non-null `site` field and point their `location` field to "Site" `LocationType` locations that have the same names and other attributes as their respective `Sites`.
```

Comment:
Let's be cautious about using controlled vocabulary. The data migration must be created manually, but it's actually a data migration, so I think it would be better to describe that and then use "data migration" consistently.

```suggestion
Now we can write the function that does the data migration. We need to query `ExampleModel` instances that have non-null `site` field and point their `location` field to "Site" `LocationType` locations that have the same names and other attributes as their respective `Sites`.
```